lib/common/attrd_client.c

```diff
@@ -293,13 +288,15 @@ pcmk__node_attr_target(const char *name)
 
         // It is important to use the name by which the scheduler knows us
         if (host_physical && pcmk__str_eq(target, "host", pcmk__str_casei)) {
-            name = host_physical;
+            free(name);
+            return strdup(host_physical);
```

Comment:
I think it's nonintuitive that this would free its argument. You'll also have a problem if the argument was allocated by glib.

I'm thinking have the function return a const char *, but return NULL if name should be used as-is. That way the caller controls how the result is allocated/freed.